Dockerfile

```diff
@@ -7,19 +7,29 @@ ARG gid=1000
 ARG http_port=8080
 ARG agent_port=50000
 
-ENV JAVA_OPTS -Djava.awt.headless=true
-ENV JENKINS_HOME /var/jenkins_home
+# Note: new env var, specific to Essentials. Not strictly necessary, but will help avoid
+# hardcode this path everywhere
+# *NOT* /var/jenkins_home on purpose, because we probably want people to ack the structure is
+# not the same as the jenkins/jenkins image
+ENV JENKINS_DIR /var/jenkins
+ENV JENKINS_HOME ${JENKINS_DIR}/home
 ENV JENKINS_AGENT_PORT ${agent_port}
 ENV EVERGREEN_ENDPOINT=http://127.0.0.1:9292
 ENV COPY_REFERENCE_FILE_LOG $JENKINS_HOME/copy_reference_file.log
 ENV JENKINS_UC https://updates.jenkins.io
 ENV JENKINS_UC_EXPERIMENTAL=https://updates.jenkins.io/experimental
 
-RUN mkdir -p /usr/share/jenkins/ref/init.groovy.d
+ENV JAVA_OPTS="-Djava.awt.headless=true -Djenkins.model.Jenkins.WORKSPACES_DIR=${JENKINS_DIR}/var/\${ITEM_FULL_NAME}/workspace -Djenkins.model.Jenkins.BUILDS_DIR=$JENKINS_DIR/var/\${ITEM_FULL_NAME}/builds"
```

Comment:
the `jenkins.model.Jenkins.*_DIR` won't do anything until https://github.com/jenkinsci/jenkins/pull/3360 or some form of it gets merged and we can use the tip of the core.